src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingNullCaseInSwitchCheck.java

```diff
@@ -0,0 +1,205 @@
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code missing.switch.nullcase}
+ * </li>
+ * </ul>
+ *
+ * @since 10.18.0
+ */
+
+@StatelessCheck
+public class MissingNullCaseInSwitchCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "missing.switch.nullcase";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.LITERAL_SWITCH};
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final List<DetailAST> caseLabels = getAllCaseLabels(ast);
+        final boolean hasNullCaseLabel = caseLabels.stream()
+                .anyMatch(MissingNullCaseInSwitchCheck::hasLiteralNull);
+        if (!hasNullCaseLabel) {
+            final boolean hasPatternCaseLabel = caseLabels.stream()
+                .anyMatch(MissingNullCaseInSwitchCheck::hasPatternCaseLabel);
+            final boolean hasStringCaseLabel = caseLabels.stream()
+                .anyMatch(MissingNullCaseInSwitchCheck::hasStringCaseLabel);
+            if (hasPatternCaseLabel || hasStringCaseLabel) {
+                log(ast, MSG_KEY);
+            }
+        }
+    }
+
+    /**
+     * Gets all case labels in the given switch AST node.
+     *
+     * @param detailAST the AST node representing {@code LITERAL_SWITCH}
+     * @return a list of all case labels in the switch
+     */
+    private static List<DetailAST> getAllCaseLabels(DetailAST detailAST) {
+        final List<DetailAST> caseLabels = new ArrayList<>();
+        DetailAST ast = detailAST.getFirstChild();
+        while (ast != null) {
+            // case group token may have several LITERAL_CASE tokens
+            TokenUtil.forEachChild(ast, TokenTypes.LITERAL_CASE, caseLabels::add);
+            ast = ast.getNextSibling();
+        }
+        return Collections.unmodifiableList(caseLabels);
+    }
+
+    /**
+     * Checks if the given case AST node has a null label.
+     *
+     * @param caseLabel the AST node representing {@code LITERAL_CASE}
+     * @return true if the case has {@code null} label, false otherwise
+     */
+    private static boolean hasLiteralNull(DetailAST caseLabel) {
+        return Optional.ofNullable(caseLabel.findFirstToken(TokenTypes.EXPR))
+                .map(exp -> exp.findFirstToken(TokenTypes.LITERAL_NULL))
+                .isPresent();
+    }
+
+    /**
+     * Checks if the given case AST node has a pattern variable declaration label.
+     * or record pattern definition label.
+     * In this case, if the switch selector evaluates to null,
+     * a {@code NullPointerException} will be thrown.
```

Comment:
This sentence doesn't seem to have anything to do with the method. This seems to be more about the check overall. The method javadoc should be specific to what the implementation of the method is doing to convey this to the user. We are just checking if this has a pattern case. The reason for **why** we are checking it is more for the caller and it's logic.

I would remove it or move it somewhere more pertinent.

Reply:
Yeah, let’s just remove it

Reply:
done